apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java

```diff
@@ -131,20 +136,43 @@ public boolean canExpireSnapshots(TableMetadata tableMetadata) {
   public boolean canRunRetention(TableMetadata tableMetadata) {
     GetTableResponseBody response = getTable(tableMetadata);
 
-    if (response == null || !isPrimaryTable(response)) {
+    if (response == null || !isOlderTable(response) || !isPrimaryTable(response)) {
       return false;
     }
     Optional<RetentionConfig> config = getTableRetention(response);
     return config.isPresent();
   }
 
-  private boolean isPrimaryTable(@NonNull GetTableResponseBody response) {
-    return GetTableResponseBody.TableTypeEnum.PRIMARY_TABLE == response.getTableType();
+  /**
+   * Checks if staged deletion task can be run on given table
+   *
+   * @param tableMetadata
+   * @return
+   */
+  public boolean canRunStagedDataDeletion(@NonNull TableMetadata tableMetadata) {
+    GetTableResponseBody response = getTable(tableMetadata);
+    return response != null && isOlderTable(response);
   }
 
-  private boolean isPrimaryTable(@NonNull TableMetadata tableMetadata) {
+  /**
+   * Checks if orphan files deletion task can be run on given table
+   *
+   * @param tableMetadata
+   * @return
+   */
+  public boolean canRunOrphanFileDeletion(@NonNull TableMetadata tableMetadata) {
     GetTableResponseBody response = getTable(tableMetadata);
-    return response != null && isPrimaryTable(response);
+    return response != null && isOlderTable(response);
+  }
+
+  private boolean isOlderTable(@NonNull GetTableResponseBody response) {
+    return response.getCreationTime() != null
```

Comment:
so those tables with creationTime missing will not be picked by any jobs ?

Reply:
The `GetTableResponseBody` is from `getTable` api NOT tables/search api. So it should all have creationTime. Even if it returns default int value (0), the table should be picked up.

Reply:
From a code readability standpoint can we make the logic easy to follow. I don't know the assumption here of value being 0 if creationTime is null. 

if (creationTime == null) {
   // no creation time for the table, by default we pick these tables for maintenance.
   return true;
} else {
   ....
}

Reply:
Addressed. Agree, this adds more clarity.